### PR TITLE
Add Missing Examples

### LINKED ISSUE #386: [Feature]: Examples coverage — one compilable example per module (pilot M3)
Part of #375. Examples are the SSOT for integration/composition. Add one compilable, CI-tested example per module (the `rate_limiter` examples are the model). Pilot the M3 modules first, then backfill:

- [ ] `access_control` (folds the closed #378)
- [ ] `two_step_transfer`
- [ ] `delayed_transfer`
- [ ] math modules (as recipes warrant)

**Done:** each listed module has an example package under its `examples/` dir, compiled + tested in CI.

### PR DESCRIPTION
**Pull request by ericnordelo**
Resolves #386  

## Summary by CodeRabbit

* **New Features**
  * Added new example modules for role-gated reward treasuries, timelocked treasury withdrawals, and two-step operator capability handoff.
  * Added math examples including token normalization, AMM quoting, fee splitting, median price oracles, z-score analytics, and fixed-point compounding.
* **Documentation**
  * Updated package READMEs to add richer “Examples” sections and links to fully compilable sample code (with unaudited callouts where applicable).
  * Extended example references in math and contracts package tables.
* **Tests**
  * Added comprehensive test coverage for success, edge cases, rounding behavior, governance/timelock flows, and expected failures.

### CODE REVIEW

**Review by 0xNeshi — APPROVED** [review]
:fire:

File: contracts/access/examples/access_control/reward_treasury.move [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+///
+/// The tests also exercise the timelocked root-admin handoff (`begin`/`accept` transfer
+/// and renounce), the integration story for rotating or retiring protocol governance.
+///
+/// # Disclaimer
+///
+/// This module is an **unaudited example**, provided purely to illustrate ways the
+/// `access_control` primitive can be integrated. It is not production-ready and must not
+/// be deployed as-is.
+module openzeppelin_access::example_reward_treasury;
```
Comment (0xNeshi) [inline contracts/access/examples/access_control/reward_treasury.move:43]: nit: Having the `example_` prefix in the module names is fine, but in the file name it seems redundant, given that the file lives in the `examples/*` dir. 

I think just `examples/access_control/reward_treasury.move` is a good name _for the file_ (again, the actual module name is fine as-is). Same for all other examples.

File: math/core/examples/decimal_scaling/token_normalizer.move [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+///
+/// # Disclaimer
+///
+/// This module is an **unaudited example**, provided purely to illustrate ways the
+/// `decimal_scaling` primitive can be integrated. It is not production-ready and must not
+/// be deployed as-is.
+module openzeppelin_math::example_token_normalizer;
+
+use openzeppelin_math::decimal_scaling;
+use sui::event;
```
[the remark this answers is not in the document]
Comment (bidzyyys) [reply math/core/examples/decimal_scaling/token_normalizer.move:40]: @coderabbitai fix it please.

**Review by bidzyyys — APPROVED** [review]
LGTM

File: math/core/examples/decimal_scaling/token_normalizer.move (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericnordelo) [reply math/core/examples/decimal_scaling/token_normalizer.move:40]: No file in the entire repo uses that header, that section in the STYLEGUIDE refers to formatting the imports, not adding that comment.
